packages/overlay/test/overlay.spec.ts

```diff
@@ -115,4 +115,22 @@ describe('overlay fs', () => {
     expect(readdirSync(srcPath, { withFileTypes: true })).to.have.lengthOf(1);
     expect(await readdir(srcPath, { withFileTypes: true })).to.have.lengthOf(1);
   });
+
+  it(`resolves real path when given the parent dir of the base dir`, async function () {
+    const lower = createMemoryFs({
+      '/src/a': {},
```

Comment:
I'd change lower to only have /src. upper should add /src/a on top of it, so it makes sense.